### PR TITLE
Generator for use with Pipelines

### PR DESCRIPTION
**Pull request by riedgar-ms**
Add a class to `generators.py` to use a Hugging Face `Pipeline` object as a generator (rather than extracting the model and tokeniser). This is @Nking92 's work, with some tweaks to get it to fit into the latest iteration of the file.

### CODE REVIEW

**Comment by riedgar-ms** [comment]
Ping @slundberg @Harsha-Nori @Nking92

**Comment by Nking92** [comment]
Thank you for finishing this up. This looks great to me. I was able to run the Adatest UI as well as the tests without any major issues. The only problem I had was needing to pip install appdirs and pytest-mock, but I see your other PR has those added to the setup.py file.

Thank you for writing nice tests. They made it much easier to understand how the generators are expected to be used.

**Comment by riedgar-ms** [comment]
Thanks @Nking92 .... @slundberg @Harsha-Nori ... are you happy to sign off and merge?

File: adatest/generators.py [hunk 1]
```diff
@@ -153,33 +153,39 @@ def _parse_suggestion_texts(self, suggestion_texts, prompts):
             #     samples.append(tuple(suggestion))
         return list(set(samples))
 
+class  HuggingFace(TextCompletionGenerator):
```
Comment (Harsha-Nori) [inline adatest/generators.py:156]: Extra space after class definition

File: tests/test_generators.py [hunk 2]
```diff
@@ -20,8 +22,97 @@ def test_smoke(self, model_name):
...
+        for item in results:
+            assert isinstance(item, str)
+
+
+GENERATOR_PIPELINE_MODELS = [
+    "facebook/opt-125m",
+    "facebook/opt-350m",
+    "EleutherAI/gpt-neo-125M",
+    "gpt2",
+    "bigscience/bloom-560m",
```
Comment (Harsha-Nori) [inline tests/test_generators.py:66]: Nice set of models to test. Just want to make sure the GH actions machines we use will consistently be able to load and use these

File: adatest/generators.py [hunk 3]
```diff
@@ -226,6 +232,45 @@ def prepare_inputs_for_generation(input_ids, **kwargs):
...
+            self._sep_stopper.prompt_length = prompt_length
+            self._sep_stopper.max_length = max_length
+            generations = self.source(p,
+                        do_sample=True,
+                        max_length=prompt_length + max_length,
+                        num_return_sequences=num_samples,
+                        pad_token_id=self.source.model.config.eos_token_id,
+                        stopping_criteria=[self._sep_stopper])
+            for gen in generations:
+                generated_text: str = gen['generated_text'][len(p):]
```
Comment (Harsha-Nori) [inline adatest/generators.py:261]: I think we can skip this type annotation on this inner variable, right? Unless it's a particularly tricky output

File: adatest/generators.py (hunk 1 above)
Comment (riedgar-ms) [reply adatest/generators.py:156]: Removed

File: adatest/generators.py (hunk 3 above)
Comment (riedgar-ms) [reply adatest/generators.py]: Removed.

File: tests/test_generators.py (hunk 2 above)
Comment (riedgar-ms) [reply tests/test_generators.py:66]: I doubt that the build cluster has GPUs, so these probably just all run on the CPU anyway?

**Review by Harsha-Nori — APPROVED** [review]
Looks good -- thanks for enabling support for this, @riedgar-ms (and @Nking92) !

**Pull request merged**